src/client.ts

```diff
@@ -62,50 +61,14 @@ class Client extends EventEmitter implements IClient {
 
     private ready: boolean = false;
 
-    constructor(config: IProxyConfig, init: Function = initialize) {
```

Comment:
I'm not a fan of the partial injection inside a constructor. It should be a responsibility of a composition root (in our case it's create-client.ts)